### ISSUE TITLE
BaseALT + LDAP + UDSactor Trouble

### ISSUE THREAD

**Issue by Rabb1a** [VirtualCable/openuds#118]
Good afternoon,

I'd like to describe an issue I'm facing. I have a Proxmox virtualization environment where OpenUDS is installed, all running on the BaseALT operating system.

The core issue is that I have a golden image of the operating system with OpenUDS Agent and XRDP installed. Additionally, there is a domain controller in the virtualization environment based on Kerberos and Samba.

After registering the agent in the system and enabling the udsactor, I shut down the machine. Then, I create a pool and bind it to the Active Directory OS manager.

During the creation of the virtual machine, it reboots, receives the name "Work-000", and I can see that the machine is registered in LDAP. However, inside the machine, it receives the name "Work-000" instead of "Work-000.domain.alt", and it doesn't recognize any domain groups.

When I check if the machine is part of the domain, it returns an error.
`[root@WorkPC-000 ~]# net ads testjoin
Join to domain is not valid: NT code 0xfffffff6
`
**Journalctl:**
`окт 01 15:26:04 WorkPC-002 dbus-daemon[2367]: [system] Activating via systemd: service name='org.freedesktop.hostname1' unit='dbus-org.freedesktop.hostname1.service' requested by ':1.106' (uid=0 pid=3410 comm="/usr/bin/hostnamectl set-hostname WorkPC-002")
окт 01 15:26:04 WorkPC-002 systemd[1]: Starting Hostname Service...
окт 01 15:26:04 WorkPC-002 dbus-daemon[2367]: [system] Successfully activated service 'org.freedesktop.hostname1'
окт 01 15:26:04 WorkPC-002 systemd[1]: Started Hostname Service.
окт 01 15:26:04 WorkPC-002 NetworkManager[2470]: <info>  [1727785564.8059] hostname: static hostname changed from "client" to "WorkPC-002"
окт 01 15:26:04 WorkPC-002 dbus-daemon[2367]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service' requested by ':1.8' (uid=0 pid=2470 comm="/usr/sbin/NetworkManager --no-daemon")
окт 01 15:26:04 WorkPC-002 systemd[1]: Starting Network Manager Script Dispatcher Service...
окт 01 15:26:04 WorkPC-002 dbus-daemon[2367]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
окт 01 15:26:04 WorkPC-002 systemd[1]: Started Network Manager Script Dispatcher Service.
окт 01 15:26:04 WorkPC-002 dbus-daemon[2367]: [system] Activating via systemd: service name='org.freedesktop.realmd' unit='realmd.service' requested by ':1.109' (uid=0 pid=3414 comm="realm join -U administrator@domain.alt --client-so")
окт 01 15:26:04 WorkPC-002 systemd[1]: Starting Realm and Domain Configuration...
окт 01 15:26:04 WorkPC-002 realmd[3423]: Loaded settings from: /usr/lib/realmd/realmd-defaults.conf /usr/lib/realmd/realmd-distro.conf /etc/realmd.conf
окт 01 15:26:04 WorkPC-002 realmd[3423]: holding daemon: startup
окт 01 15:26:04 WorkPC-002 realmd[3423]: starting service
окт 01 15:26:04 WorkPC-002 realmd[3423]: connected to bus
окт 01 15:26:04 WorkPC-002 realmd[3423]: GLib-GIO: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
окт 01 15:26:04 WorkPC-002 realmd[3423]: released daemon: startup
окт 01 15:26:04 WorkPC-002 dbus-daemon[2367]: [system] Successfully activated service 'org.freedesktop.realmd'
окт 01 15:26:04 WorkPC-002 systemd[1]: Started Realm and Domain Configuration.
окт 01 15:26:04 WorkPC-002 realmd[3423]: claimed name on bus: org.freedesktop.realmd
окт 01 15:26:04 WorkPC-002 realmd[3423]: client using service: :1.109
окт 01 15:26:04 WorkPC-002 realmd[3423]: holding daemon: :1.109
окт 01 15:26:04 WorkPC-002 realmd[3423]: Using 'r6208.3414' operation for method 'Discover' invocation on 'org.freedesktop.realmd.Provider' interface
окт 01 15:26:04 WorkPC-002 realmd[3423]: Registered cancellable for operation 'r6208.3414'
окт 01 15:26:04 WorkPC-002 realmd[3423]:  * Resolving: _ldap._tcp.domain.alt
окт 01 15:26:04 WorkPC-002 realmd[3423]:  * Performing LDAP DSE lookup on: 10.61.24.2
окт 01 15:26:04 WorkPC-002 realmd[3423]: Searching  for (objectClass=*)
окт 01 15:26:04 WorkPC-002 realmd[3423]: Got defaultNamingContext: DC=domain,DC=alt
окт 01 15:26:04 WorkPC-002 realmd[3423]: Sending TCP Netlogon request
окт 01 15:26:04 WorkPC-002 realmd[3423]: Received TCP Netlogon response
окт 01 15:26:04 WorkPC-002 realmd[3423]:  * Successfully discovered: domain.alt
окт 01 15:26:04 WorkPC-002 realmd[3423]: Using 'r6208.3414' operation for method 'Join' invocation on 'org.freedesktop.realmd.KerberosMembership' interface
окт 01 15:26:04 WorkPC-002 realmd[3423]: Registered cancellable for operation 'r6208.3414'
окт 01 15:26:04 WorkPC-002 realmd[3423]: holding daemon: current-invocation
окт 01 15:26:04 WorkPC-002 realmd[3423]:  * Required files: /usr/sbin/sss_cache, /usr/libexec/sssd/sssd_pac, /usr/libexec/sssd/sssd_ifp, /usr/sbin/sssd, /usr/libexec/sssd/gpo_child, /usr/bin/net
окт 01 15:26:04 WorkPC-002 realmd[3423]:  * LANG=C LOGNAME=root /usr/bin/net --configfile /var/cache/realmd/realmd-smb-conf.ZINPU2 -U administrator@domain.alt --use-kerberos=required ads join domain.alt createcomputer=WORKPC
окт 01 15:26:04 WorkPC-002 realmd[3423]: process started: 3471
окт 01 15:26:06 WorkPC-002 realmd[3423]: Password for [administrator@domain.alt]:DNS update failed: NT_STATUS_INVALID_PARAMETER
окт 01 15:26:06 WorkPC-002 realmd[3423]:
окт 01 15:26:06 WorkPC-002 realmd[3423]: Using short domain name -- DOMAIN
окт 01 15:26:06 WorkPC-002 realmd[3423]: Joined 'WORKPC-002' to dns domain 'domain.alt'
окт 01 15:26:06 WorkPC-002 realmd[3423]: No DNS domain configured for workpc-002. Unable to perform DNS Update.
окт 01 15:26:06 WorkPC-002 realmd[3423]: process exited: 3471
окт 01 15:26:06 WorkPC-002 realmd[3423]:  * LANG=C LOGNAME=root /usr/bin/net --configfile /var/cache/realmd/realmd-smb-conf.ZINPU2 -U administrator@domain.alt ads keytab create
окт 01 15:26:06 WorkPC-002 realmd[3423]: process started: 3475
окт 01 15:26:07 WorkPC-002 realmd[3423]: Password for [administrator@domain.alt]:
окт 01 15:26:07 WorkPC-002 realmd[3423]: process exited: 3475
окт 01 15:26:07 WorkPC-002 realmd[3423]:  * /sbin/chkconfig sssd on
окт 01 15:26:07 WorkPC-002 realmd[3423]: process started: 3477
окт 01 15:26:07 WorkPC-002 realmd[3423]: Внимание: Отправляется запрос 'systemctl enable sssd.service'.
окт 01 15:26:07 WorkPC-002 realmd[3423]: Synchronizing state of sssd.service with SysV service script with /lib/systemd/systemd-sysv-install.
окт 01 15:26:07 WorkPC-002 realmd[3423]: Executing: /lib/systemd/systemd-sysv-install enable sssd
окт 01 15:26:08 WorkPC-002 systemd[1]: Reloading.
окт 01 15:26:08 WorkPC-002 systemd-sysv-generator[3497]: SysV service '/etc/rc.d/init.d/pppoe' lacks a native systemd unit file. Automatically generating a unit file for compatibility. Please update package to include a native systemd unit file, in order to make it more safe and robust.
окт 01 15:26:08 WorkPC-002 systemd-sysv-generator[3497]: SysV service '/etc/rc.d/init.d/clock' lacks a native systemd unit file. Automatically generating a unit file for compatibility. Please update package to include a native systemd unit file, in order to make it more safe and robust.
окт 01 15:26:08 WorkPC-002 systemd[1]: /lib/systemd/system/cups.socket:6: ListenStream= references a path below legacy directory /var/run/, updating /var/run/cups/cups.sock → /run/cups/cups.sock; please update the unit file accordingly.
окт 01 15:26:08 WorkPC-002 realmd[3423]: Created symlink /etc/systemd/system/multi-user.target.wants/sssd.service → /lib/systemd/system/sssd.service.
окт 01 15:26:08 WorkPC-002 systemd[1]: Reloading.
окт 01 15:26:08 WorkPC-002 systemd-sysv-generator[3513]: SysV service '/etc/rc.d/init.d/pppoe' lacks a native systemd unit file. Automatically generating a unit file for compatibility. Please update package to include a native systemd unit file, in order to make it more safe and robust.
окт 01 15:26:08 WorkPC-002 systemd-sysv-generator[3513]: SysV service '/etc/rc.d/init.d/clock' lacks a native systemd unit file. Automatically generating a unit file for compatibility. Please update package to include a native systemd unit file, in order to make it more safe and robust.
окт 01 15:26:08 WorkPC-002 systemd[1]: /lib/systemd/system/sssd.service:17: PIDFile= references a path below legacy directory /var/run/, updating /var/run/sssd.pid → /run/sssd.pid; please update the unit file accordingly.
окт 01 15:26:08 WorkPC-002 systemd[1]: /lib/systemd/system/cups.socket:6: ListenStream= references a path below legacy directory /var/run/, updating /var/run/cups/cups.sock → /run/cups/cups.sock; please update the unit file accordingly.
окт 01 15:26:08 WorkPC-002 realmd[3423]: process exited: 3477
окт 01 15:26:08 WorkPC-002 realmd[3423]:  * /sbin/service sssd restart
окт 01 15:26:08 WorkPC-002 realmd[3423]: process started: 3515
окт 01 15:26:08 WorkPC-002 systemd[1]: Starting System Security Services Daemon...
окт 01 15:26:09 WorkPC-002 sssd[3518]: Starting up
окт 01 15:26:09 WorkPC-002 sssd_be[3519]: Starting up
окт 01 15:26:09 WorkPC-002 sssd_pac[3522]: Starting up
окт 01 15:26:09 WorkPC-002 sssd_nss[3520]: Starting up
окт 01 15:26:09 WorkPC-002 sssd_pam[3521]: Starting up
окт 01 15:26:09 WorkPC-002 systemd[1]: Started System Security Services Daemon.
окт 01 15:26:09 WorkPC-002 realmd[3423]: process exited: 3515
окт 01 15:26:09 WorkPC-002 realmd[3423]:  * /usr/lib/realmd/realmd-altlinux-helper enable
окт 01 15:26:09 WorkPC-002 realmd[3423]: process started: 3524
окт 01 15:26:09 WorkPC-002 realmd[3423]: groupadd: группа «localadmins» уже существует
окт 01 15:26:09 WorkPC-002 systemd[1]: Reexecuting.
окт 01 15:26:09 WorkPC-002 systemd[1]: systemd v249.17-alt2 running in system mode (+PAM +AUDIT +SELINUX -APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS -FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 +PWQUALITY -P11KIT -QRENCODE +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified)
окт 01 15:26:09 WorkPC-002 systemd[1]: Detected virtualization kvm.
окт 01 15:26:09 WorkPC-002 systemd[1]: Detected architecture x86-64.
окт 01 15:26:09 WorkPC-002 systemd-sysv-generator[3580]: SysV service '/etc/rc.d/init.d/pppoe' lacks a native systemd unit file. Automatically generating a unit file for compatibility. Please update package to include a native systemd unit file, in order to make it more safe and robust.
окт 01 15:26:09 WorkPC-002 systemd-sysv-generator[3580]: SysV service '/etc/rc.d/init.d/clock' lacks a native systemd unit file. Automatically generating a unit file for compatibility. Please update package to include a native systemd unit file, in order to make it more safe and robust.
окт 01 15:26:10 WorkPC-002 systemd[1]: /lib/systemd/system/sssd.service:17: PIDFile= references a path below legacy directory /var/run/, updating /var/run/sssd.pid → /run/sssd.pid; please update the unit file accordingly.
окт 01 15:26:10 WorkPC-002 systemd[1]: /lib/systemd/system/cups.socket:6: ListenStream= references a path below legacy directory /var/run/, updating /var/run/cups/cups.sock → /run/cups/cups.sock; please update the unit file accordingly.
окт 01 15:26:10 WorkPC-002 realmd[3423]: process exited: 3524
окт 01 15:26:10 WorkPC-002 realmd[3423]:  * Successfully enrolled machine in realm
окт 01 15:26:10 WorkPC-002 realmd[3423]: released daemon: current-invocation
окт 01 15:26:10 WorkPC-002 realmd[3423]: client gone away: :1.109
окт 01 15:26:10 WorkPC-002 realmd[3423]: released daemon: :1.109
окт 01 15:26:10 WorkPC-002 systemd[1]: Stopping Session 7 of User user...
окт 01 15:26:10 WorkPC-002 systemd[1]: Removed slice Slice /system/getty.`

**udsactor.log:**
ERROR 2024-10-01 15:44:13,601 Error registering on actor service: HTTPSConnectionPool(host='127.0.0.1', port=43910): Max retries exceeded with url: /ui/register (Caused by NewConnectionError('<urllib3.conn
ection.HTTPSConnection object at 0x7fc6e2d08eb0>: Failed to establish a new connection: [Errno 111] В соединении отказано'))
ERROR 2024-10-01 15:44:13,992 Error on client loop: HTTPSConnectionPool(host='127.0.0.1', port=43910): Max retries exceeded with url: /ui/login (Caused by NewConnectionError('<urllib3.connection.HTTPSConne
ction object at 0x7fc6d9dc1340>: Failed to establish a new connection: [Errno 111] В соединении отказано'))
ERROR 2024-10-01 15:44:13,997 Error unregistering on actor service: HTTPSConnectionPool(host='127.0.0.1', port=43910): Max retries exceeded with url: /ui/unregister (Caused by NewConnectionError('<urllib3.
connection.HTTPSConnection object at 0x7fc6d9dc16d0>: Failed to establish a new connection: [Errno 111] В соединении отказано'))

There is very little information online, so could any of the developers possibly get in touch to provide guidance?
If u can write me in Telegram: Rabbia1

**Comment by dkmstr** [comment]
Hi... The AD join part for linux of UDS was developed by Alexander, @Future998 . 
I have never used nor tested that part, simple check there was no flaws on it. Maybe alexander can help you if you can reach him.

Anyway, what version uf UDS, UDSActor, etc.. are you using?

**Comment by Rabb1a** [comment]
[root@vdiserver ~]# rpm -qa | grep uds
openuds-server-3.6.0-alt7.noarch
**AND**
[user@client ~]$ rpm -qa | grep uds
openuds-actor-3.6.0-alt4.noarch
It would be nice to contact Alexander of course.

**Comment by Future998** [comment]
I read your issue and realized that you have two problems: 
1. You want your hostname to be like FQDN. In this case, I can say that hostname and FQDN are not the same things for Samba. In order for the VM template to work correctly with your domain, you need to configure `resolv.conf` in your VM template.
2. You have a problem with group policies. To solve this problem, you should configure your VM template according to your requirements. You can read more about this [here](https://www.altlinux.org/%D0%93%D1%80%D1%83%D0%BF%D0%BF%D0%BE%D0%B2%D1%8B%D0%B5_%D0%BF%D0%BE%D0%BB%D0%B8%D1%82%D0%B8%D0%BA%D0%B8) or [here](https://docs.altlinux.org/ru-RU/alt-education-e2k/10.1/html/alt-education-e2k/ch46.html), and you can also pay attention to other information available on the Internet.

To set up a minimal working VM template that works with domain OS managers on ALT Linux, you need to follow four simple steps:
1. Configure `resolv.conf` to interact with the domain. Namely, specify the domain to which the login will be made, as well as the DC IP address.
For example:
```
domain alt.domain
nameserver 192.168.0.1
```
The first line is the domain to be logged in,
The second is IP DC.

2. Choose whether to explicitly or implicitly specify the domain in the login username.
There are two configuration options:
2.1. Configure `/etc/realmd.conf`, namely, change the `fully-qualified-names` parameter to the desired one.
More detailed:
```> man realmd.conf```
2.2. In the settings of the Samba authenticator, on the `LDAP information` tab, specify `userPrincipalName` instead of `sAMAccountName` in the User Attribute Identifier parameter. In this case, the login will be done by the user `testuser@alt.domain`, not `testuser`.

3. Setting up the correct operation of XRDP with the domain user after entering the machine into the domain (relevant only for XRDP):
Add lines to `/etc/sssd/sssd.conf`:
```
[domain/alt.domain]
ad_gpo_map_interactive = +xrdp-sesman
```
In `/etc/xrdp/sesman.ini`, change the lines:
```
TerminalServerUsers=domain users
TerminalServerAdmins=domain users
```

4. Depersonalization of the VM:
```
> vim /etc/machine-id
> rm -fr /var/lib/dbus
> vim /etc/hostname
> vim /etc/sysconfig/network
> cd /etc/openssh/
* deleting some files (note which files are left) *
> ll
total 588
drwxr-x--x 4 root root 4096 aug 24 16:41 ./
drwxr-xr-x 174 root root 12288 aug 24 16:40 ../
drwxr-x--x 2 root root 4096 aug 13 15:23 authorized_keys/
-rw------- 1 root root 3817 aug 13 15:09 sshd_config
-rw------- 1 root root 565252 aug 13 15:09 moduli
drwxr-x--x 2 root root 4096 aug 13 15:09 authorized_keys2/
-rw-r--r-- 1 root root 1727 aug 13 15:04 ssh_config
```
Vim needs to delete the machine-id and hostname
This step must be performed last. It is necessary in order to be able to add multiple domain machines, rather than one.

With each subsequent revision of the VM template, you must repeat step 4. Depersonalization of the VM.

To summarize, I can say that this is not an OpenUDS problem, but a VM template, configure this and your problem will be solved.
If you still have questions, I can write to you by telegram.

**Comment by Rabb1a** [comment]
You are a holy man @Future998 , thank you for answer. But I left Samba and installed FreeIPA and was able to enter the PC into the domain without any problems. But now there's another problem. This is an Authenticator. Of course, if you have the opportunity to write in a telegram. It would be easier. Moreover, it is more convenient to use the same language.

![image](https://github.com/user-attachments/assets/601ef96a-0ac0-4e0d-89f3-73aaf271c190)
![image](https://github.com/user-attachments/assets/66e64353-96b9-4f71-95ff-d39ecb40e04e)
After saving, the group disappears. 
Accordingly, it is not assigned to the group, and therefore does not allow it into the system.

**Comment by Future998** [comment]
After saving, some fields are not displayed in the Web UI, but they are there - this is normal, as far as I remember.

**Comment by dkmstr** [comment]
Yes, the group is not assigned from UDS, its got from the authenticator. If you use an LDAP based instalation, try the regex LDAP, although i think the Active Directory one should work.

In order to apper the user groups, you must add first that group to UDS. (So, if you have for example an user "A" with Groups "A", "B" y "C", but none of them is registered in UDS, the groups field of UDS will appear empty (even if you mark anythin. The marks are ignored on most auths, except internal).

Thanks a lot to @Future998 For his help... :)

**Comment by Rabb1a** [comment]
Now we have User (baa) 
![image](https://github.com/user-attachments/assets/036d3155-1ad9-4082-87cd-71a84f64cd45)
**memberof: cn=altusers,cn=groups,cn=accounts,dc=ipa,dc=alt**
Setting for Regex :
![image](https://github.com/user-attachments/assets/92821bfe-d4c2-4f6e-be54-ead2f11a2cfc)
OK after, we make Group (altusers) : 
![image](https://github.com/user-attachments/assets/c44bed24-2e64-46a5-b8af-df6021ee6d02)
and Create new user (select-> baa)
![image](https://github.com/user-attachments/assets/c68f0343-6ac6-4717-b8a6-45c632f4524b)
Look Info about baa > And he dont have group.
After we try Connect to UDS with Credationals

![image](https://github.com/user-attachments/assets/57af95c0-2f40-4de3-afbf-5e14503bc3c7)
and go to he Log. 

![image](https://github.com/user-attachments/assets/4b5f1f80-c0b7-4afa-94e1-7b955b6cf94b)

Doesn't work

**Comment by Rabb1a** [comment]
RESOLVE:
![image](https://github.com/user-attachments/assets/39cb9cdd-3630-4a37-bcc3-532a29b968f6)
and dont need ADD USER. After Login in UDS, UDS create User auto.
thx.
